SportsHub/SportsHub.Api/Controllers/SportController.cs

```diff
@@ -0,0 +1,131 @@
+﻿using AutoMapper;
+using FluentValidation;
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SportsHub.Api.Mapping.Models;
+using SportsHub.Api.Validations;
+using SportsHub.AppService.Authentication.Models.DTOs;
+using SportsHub.AppService.Services;
+using SportsHub.Domain.Constants;
+
+namespace SportsHub.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class SportController : ControllerBase
+    {
+        private readonly IValidator<CreateSportDTO> _sportValidator;
+        private readonly ISportService _sportService;
+        private readonly IMapper _mapper;
+
+        public SportController(IValidator<CreateSportDTO> sportValidator,
+                IGenerateModelStateDictionary generateModelStateDictionary,
+                ISportService sportService,
+                IMapper mapper)
+        {
+            _sportValidator = sportValidator;
+            _sportService = sportService;
+            _mapper = mapper;
+        }
+
+        [HttpGet("GetAllSports")]
+        public async Task<ActionResult<IEnumerable<SportResponseDTO>>> GetAllAsync()
+        {
+            var sports = await _sportService.GetAllAsync();
+
+            if (!sports.Any())
+            {
+                return BadRequest(ValidationMessages.NoSports);
```

Comment:
@MBocharskyi suggested we do not return `BadRequest` when the returned collection is empty.